debian/postinst

```diff
@@ -49,10 +49,17 @@ configure)
 
 	systemctl enable auditd.service
 	systemctl enable delphix.target
-	systemctl enable delphix-migration.service
 	systemctl enable delphix-platform.service
 	systemctl enable systemd-networkd.service
 
+	#
+	# Only enable the delphix-migration service on initial install, not
+	# on upgrade.
+	#
+	if [[ -z "$2" ]]; then
+		systemctl enable delphix-migration.service
```

Comment:
Did you consider removing this enable altogether, and then having a different change to appliance-build.. to enable this service via the appliance-build ansible logic or the migration image creation logic?

I don't think we ever want to enable the migration service, except when performing a migration, so it might be better to only enable it when building the migration artifact, since that will be a migration specific step.

For example, I don't think we should have migration enabled on initial boot, for a DE that started on 6.0, and thus was never migrated. As far as I understand this change, this way, we'll still start the migration service for initial 6.0 installs.

Reply:
Yeah, right now delphix-migration always starts on first boot and then disables itself after running successfully. Having the migration service enabled only for the migration artifact would probably have been cleaner, that said we'll be already shipping 6.0.0.0 with the current behaviour, so if we were to change this we'd then have differing behaviours depending on the version.

Looking at appliance-build, we currently use the same `binary` directory for generating all the artifacts, so we'd need to make a separate copy of the directory (in a similar way that `90-raw-disk-image.binary` does it) with the change to enable the `delphix-migration service` there. That should work but that would require some extra testing.

Do you think we should dig into this more?

Reply:
> so if we were to change this we'd then have differing behaviours depending on the version.

I don't think we should limit improvement in new versions, simply because old versions wouldn't get that same improvement.

> Do you think we should dig into this more?

It's up to you; if this works, I think it's fine. I'll approve as-is.